include/wx/string.h

```diff
@@ -2560,6 +2570,10 @@ class WXDLLIMPEXP_BASE wxString
       if ( !ch.IsAscii() )
           m_impl.assign(wxStringOperations::EncodeNChars(n, ch));
       else
+#elif wxUSE_UNICODE_WCHAR && SIZEOF_WCHAR_T == 2
```

Comment:
Ah, sorry, I didn't realize it from looking at the diff.

I agree that the logic is more clear in the latter version. I wonder if doing something like

```cpp
#if wxUSE_UNICODE_UTF8 || wxUSE_UNICODE_UTF16
    if ( wxStringOperations::RequiresEncoding(ch) )
        m_impl.assign(wxStringOperations::EncodeNChars(n, ch));
    else
#endif
        m_impl.assign(n, (wxStringCharType)ch);
```
could be even better because it's shorter, but still -- hopefully -- quite clear?

In fact, I even wonder if we could get rid of the `#if`: if `RequiresEncoding()` is just an inlined function returning false in `wxUSE_UNICODE_WCHAR || !wxUSE_UNICODE` case, it and the branch controlled by it should be completely optimized away by the compiler anyhow...